### PR TITLE
Avoid calling /database/:id/metadata in BrowseApp

### PR DESCRIPTION
**Pull request by paulrosenzweig**
If we fetch tables with just a `dbId`, the tables entity fetches all db metadata since that's the only endpoint it can use. I updated BrowseApp to pass `schemaName` even if there's just one schema so we can load tables by schema rather than fetching everything.

### CODE REVIEW

File: frontend/src/metabase/components/BrowseApp.jsx [hunk 1]
```diff
@@ -72,7 +72,12 @@ export class SchemaBrowser extends React.Component {
                 </Grid>
               </Box>
             ) : (
-              <TableBrowser {...this.props} />
+              <TableBrowser
+                {...this.props}
+                params={{ ...this.props.params, schemaName: schemas[0].name }}
+                // hide the schema since there's only one
```
Comment (walterl) [inline frontend/src/metabase/components/BrowseApp.jsx]: Is there anything preventing `schemas.length` from being 0?

Comment (paulrosenzweig) [reply frontend/src/metabase/components/BrowseApp.jsx]: Ah yeah good call! I think that could happen if there were no tables in a db.

File: frontend/src/metabase/components/BrowseApp.jsx [hunk 2]
```diff
@@ -32,7 +32,14 @@ export class SchemaBrowser extends React.Component {
...
           {({ schemas }) =>
-            schemas.length > 1 ? (
+            schemas.length === 1 ? (
+              <TableBrowser
+                {...this.props}
+                params={{ ...this.props.params, schemaName: schemas[0].name }}
+                // hide the schema since there's only one
+                showSchemaInHeader={false}
+              />
+            ) : (
```
Comment (walterl) [inline frontend/src/metabase/components/BrowseApp.jsx]: In the `schemas.length === 0` case, will the empty `<Grid>` below display "correctly"/nicely?

**Review by walterl — APPROVED** [review]
LGTM. The comment is just to make sure.

File: frontend/src/metabase/components/BrowseApp.jsx (hunk 2 above)
Comment (paulrosenzweig) [reply frontend/src/metabase/components/BrowseApp.jsx]: It matched the previous behavior of displaying nothing (under the breadcrumb header), but showing a note for empty dbs is definitely better! I just added one.

**Pull request merged**